source/common/http/utility.cc

```diff
@@ -275,6 +275,17 @@ Utility::QueryParams Utility::parseQueryString(absl::string_view url) {
   return parseParameters(url, start);
 }
 
+Utility::QueryParams Utility::parseAndDecodeQueryString(absl::string_view url) {
+  size_t start = url.find('?');
+  if (start == std::string::npos) {
+    QueryParams params;
+    return params;
+  }
+
+  start++;
+  return parseParameters(PercentEncoding::decode(StringUtil::subspan(url, start, url.size())), 0);
```

Comment:
Oh, I apologize. I read this code incorrectly the first time. I thought we could just implement this method as:

`return PercentEncoding::decode(parseQueryString(url))`

I see now that's percent-decoding the params and then parsing them.

Doesn't this have a bug? If I pass query params with a percent-encoded `&` (e.g. `x=a%26b`) then decoding before splitting parameters will split in the wrong location, yes?

Reply:
You're right! Will fix this while waiting for the chance to use GURL (https://github.com/envoyproxy/envoy/pull/11670) to arrive.